pkg/events/event_system.go

```diff
@@ -125,3 +169,38 @@ func (ec *EventSystemImpl) AddEvent(event *si.EventRecord) {
 		metrics.GetEventMetrics().IncEventsNotChanneled()
 	}
 }
+
+func (ec *EventSystemImpl) readIsTrackingEnabled() bool {
+	return common.GetConfigurationBool(configs.GetConfigMap(), configs.CMEventTrackingEnabled, configs.DefaultEventTrackingEnabled)
+}
+
+func (ec *EventSystemImpl) readRequestCapacity() int {
+	return common.GetConfigurationInt(configs.GetConfigMap(), configs.CMEventRequestCapacity, configs.DefaultEventRequestCapacity)
+}
+
+func (ec *EventSystemImpl) readRingBufferCapacity() uint64 {
+	return common.GetConfigurationUint(configs.GetConfigMap(), configs.CMEventRingBufferCapacity, configs.DefaultEventRingBufferCapacity)
+}
+
+func (ec *EventSystemImpl) isRestartNeeded() bool {
+	ec.Lock()
+	defer ec.Unlock()
+	return ec.readIsTrackingEnabled() != ec.trackingEnabled
+}
+
+func (ec *EventSystemImpl) Restart() {
+	ec.Stop()
+	ec.StartServiceWithPublisher(true)
+}
+
+func (ec *EventSystemImpl) reloadConfig() {
+	ec.requestCapacity = ec.readRequestCapacity()
+	newRingBufferCapacity := ec.readRingBufferCapacity()
+
+	// resize the ring buffer with new capacity
+	ec.eventBuffer.Resize(newRingBufferCapacity)
```

Comment:
Here, it's useful to print a log message if the resize results in losing events.
I'd check the lowest ID before/after and then print something.

Reply:
added INFO message during resize() and DEBUG message in buffer overflow case